src/diffusion/model.py

```diff
@@ -11,10 +12,26 @@
 from src.metrics import calculate_batch_ce
 
 
-def get_components(name):
+def get_components(name: str, mode: str = "same", num_attention_heads: int = 16):
     model = AutoModel.from_pretrained(name)
+    emb_dim = model.config.d_model
+
+    if mode == "same":
+        return model.encoder, model.decoder, emb_dim
+
+    elif mode == "bert":
+        decoder_config = BertConfig(
```

Comment:
We also should consider these parameters:
- `num_hidden_layers` (default 12)
- `num_attention_heads` (default 12)
- `intermediate_size` (default 3072)

For example, we can use 3-layer transformer with 2560 feedforward dim. Also, I suggest to use the same number of attention heads as in blenderbot. It doesn't affect number of parameters